### PR TITLE
Feature: 회원탈퇴 기능 구현 완료

### LINKED ISSUE #14: Feature: 회원정보 수정 기능 구현
### Feature List

- [ ] one
- [ ] two
- [ ] three

### LINKED ISSUE #15: Feature: 회원탈퇴 기능 구현
### Feature List

- [ ] one
- [ ] two
- [ ] three

### PR DESCRIPTION
**Pull request by ghkdgus29**
## Key Feature

- 회원 탈퇴 기능을 구현하였습니다.
- 인터셉터에서 특정 경로의 특정 메서드는 항상 true 를 반환하도록 하는 WhiteList 클래스를 만들었습니다.
- JPA Auditing 클래스에서 생성자와 수정자를 제거하였습니다.
- 컨트롤러 테스트 코드의 버그를 잡았습니다.
- P6Spy 를 추가하였습니다.

## To reviewer

- 기능이 간단하여 PR이 짧지만, 잘 부탁드립니다.!

Close #14

### CODE REVIEW

**Comment by yeonise** [comment]
### Test Coverage Report
|Overall Project|95.01%|:green_apple:|
|:-|:-|:-:|
|Files changed|100%|:green_apple:|
<br>

|File|Coverage||
|:-|:-|:-:|
|[Reply.java](https://github.com/mujik-tigers/on-and-off/blob/c7d911919bc0524a4648240eeb30fc9b34a08833/src%2Fmain%2Fjava%2Fsite%2Fonandoff%2Freply%2FReply.java)|100%|:green_apple:|
|[BaseTimeEntity.java](https://github.com/mujik-tigers/on-and-off/blob/c7d911919bc0524a4648240eeb30fc9b34a08833/src%2Fmain%2Fjava%2Fsite%2Fonandoff%2Futil%2FBaseTimeEntity.java)|100%|:green_apple:|
|[Comment.java](https://github.com/mujik-tigers/on-and-off/blob/c7d911919bc0524a4648240eeb30fc9b34a08833/src%2Fmain%2Fjava%2Fsite%2Fonandoff%2Fcomment%2FComment.java)|100%|:green_apple:|
|[Tag.java](https://github.com/mujik-tigers/on-and-off/blob/c7d911919bc0524a4648240eeb30fc9b34a08833/src%2Fmain%2Fjava%2Fsite%2Fonandoff%2Ftag%2FTag.java)|100%|:green_apple:|
|[Image.java](https://github.com/mujik-tigers/on-and-off/blob/c7d911919bc0524a4648240eeb30fc9b34a08833/src%2Fmain%2Fjava%2Fsite%2Fonandoff%2Fimage%2FImage.java)|100%|:green_apple:|
|[Member.java](https://github.com/mujik-tigers/on-and-off/blob/c7d911919bc0524a4648240eeb30fc9b34a08833/src%2Fmain%2Fjava%2Fsite%2Fonandoff%2Fmember%2FMember.java)|100%|:green_apple:|
|[WebConfig.java](https://github.com/mujik-tigers/on-and-off/blob/c7d911919bc0524a4648240eeb30fc9b34a08833/src%2Fmain%2Fjava%2Fsite%2Fonandoff%2Fconfig%2FWebConfig.java)|100%|:green_apple:|
|[Post.java](https://github.com/mujik-tigers/on-and-off/blob/c7d911919bc0524a4648240eeb30fc9b34a08833/src%2Fmain%2Fjava%2Fsite%2Fonandoff%2Fpost%2FPost.java)|100%|:green_apple:|
|[MemberController.java](https://github.com/mujik-tigers/on-and-off/blob/c7d911919bc0524a4648240eeb30fc9b34a08833/src%2Fmain%2Fjava%2Fsite%2Fonandoff%2Fmember%2Fpresentation%2FMemberController.java)|100%|:green_apple:|
|[ResponseMessage.java](https://github.com/mujik-tigers/on-and-off/blob/c7d911919bc0524a4648240eeb30fc9b34a08833/src%2Fmain%2Fjava%2Fsite%2Fonandoff%2Futil%2Fapi%2FResponseMessage.java)|100%|:green_apple:|
|[MemberService.java](https://github.com/mujik-tigers/on-and-off/blob/c7d911919bc0524a4648240eeb30fc9b34a08833/src%2Fmain%2Fjava%2Fsite%2Fonandoff%2Fmember%2Fapplication%2FMemberService.java)|100%|:green_apple:|
|[Bookmark.java](https://github.com/mujik-tigers/on-and-off/blob/c7d911919bc0524a4648240eeb30fc9b34a08833/src%2Fmain%2Fjava%2Fsite%2Fonandoff%2Fbookmark%2FBookmark.java)|100%|:green_apple:|
|[WhiteList.java](https://github.com/mujik-tigers/on-and-off/blob/c7d911919bc0524a4648240eeb30fc9b34a08833/src%2Fmain%2Fjava%2Fsite%2Fonandoff%2Finterceptor%2FWhiteList.java)|100%|:green_apple:|
|[AccessTokenInterceptor.java](https://github.com/mujik-tigers/on-and-off/blob/c7d911919bc0524a4648240eeb30fc9b34a08833/src%2Fmain%2Fjava%2Fsite%2Fonandoff%2Finterceptor%2FAccessTokenInterceptor.java)|96.15%|:green_apple:|
|[Notification.java](https://github.com/mujik-tigers/on-and-off/blob/c7d911919bc0524a4648240eeb30fc9b34a08833/src%2Fmain%2Fjava%2Fsite%2Fonandoff%2Fnotification%2FNotification.java)|0%|:green_apple:|

File: src/main/java/site/onandoff/interceptor/AccessTokenInterceptor.java [hunk 1]
```diff
@@ -16,7 +17,9 @@
 public class AccessTokenInterceptor implements HandlerInterceptor {
 
 	private static final String AUTHORIZATION_HEADER = "Authorization";
-	public static final String ACCOUNT_ID = "id";
+	private static final String ACCOUNT_ID = "id";
+	private static final WhiteList whiteList = WhiteList.create()
+		.addPathAndMethod("/members", HttpMethod.POST);
```
Comment (yeonise) [inline src/main/java/site/onandoff/interceptor/AccessTokenInterceptor.java:22]: 👏🏻

File: src/main/java/site/onandoff/interceptor/WhiteList.java [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+	public WhiteList addPathAndMethod(String path, HttpMethod httpMethod) {
+		Set<String> httpMethodsOfPath = paths.computeIfAbsent(path, key -> new HashSet<>());
+		httpMethodsOfPath.add(httpMethod.name());
+		return this;
+	}
+
+	public boolean contains(String path, String httpMethod) {
+		return paths.containsKey(path) && paths.get(path).contains(httpMethod);
+	}
+
```
Comment (yeonise) [inline src/main/java/site/onandoff/interceptor/WhiteList.java:31]: 시간 복잡도까지 고려한 좋은 코드네요! 😊

File: src/test/java/site/onandoff/member/application/MemberServiceTest.java [hunk 3]
```diff
@@ -136,4 +137,33 @@ void modifyPasswordFail() {
...
+		// given
+		Member member = new Member("ghkdgus29@naver.com", "hyun", "1234567a!", Provider.LOCAL);
+		memberRepository.save(member);
+
+		// when
+		memberService.deleteMember(member.getId());
+
+		// then
+		List<Member> members = memberRepository.findAll();
+		assertThat(members.size()).isEqualTo(0);
```
Comment (yeonise) [inline src/test/java/site/onandoff/member/application/MemberServiceTest.java:152]: 🩵 `assertThat(members).isEmpty()` 로도 할 수 있어요! ㅎㅎ

File: src/main/java/site/onandoff/member/application/MemberService.java [hunk 4]
```diff
@@ -41,20 +40,28 @@ public SignUpSuccessResponse signUp(@Valid UniqueSignUpForm signUpForm) {
...
 	public ModifiedMember modifyPassword(@Valid ValidPasswordChangeForm passwordChangeForm) {
-		Member member = memberRepository.findById(passwordChangeForm.getId()).orElseThrow(MemberNotFoundException::new);
+		Member member = findMemberBy(passwordChangeForm.getId());
 		member.modifyPassword(BCryptManager.encrypt(passwordChangeForm.getNewPassword()));
 
 		return new ModifiedMember(member.getId(), member.getNickname());
 	}
 
+	public void deleteMember(Long memberId) {
+		Member member = findMemberBy(memberId);
```
Comment (yeonise) [inline src/main/java/site/onandoff/member/application/MemberService.java:58]: 💁🏻 개인적인 생각이지만, 회원가입과 관련된 네이밍을 `sign up` 으로 사용한 것처럼 회원탈퇴 관련 컨트롤러나 서비스 메서드 네이밍도 비즈니스적인 관점에서 (ex. withdraw) 작성하는건 어떻게 생각하시나요? 🤔

**Review by yeonise — COMMENTED** [review]
이번 PR도 고생하셨어요! 항상 좋은 구조를 고민하시는 것 같네요 ㅎㅎ 👍🏻

File: src/test/java/site/onandoff/member/application/MemberServiceTest.java (hunk 3 above)
Comment (ghkdgus29) [reply src/test/java/site/onandoff/member/application/MemberServiceTest.java:152]: 글쿤요~ 담엔 그렇게 해야겠네요

File: src/main/java/site/onandoff/member/application/MemberService.java (hunk 4 above)
Comment (ghkdgus29) [reply src/main/java/site/onandoff/member/application/MemberService.java:58]: resign 이나 withdraw 도 생각해보긴 했는데, Repository 명도 delete 이고, 다른 엔티티들도 삭제할때 delete 를 사용할텐데 단어를 다르게해서 복잡성을 높여야하나 고민을 좀 해봤어요!

**Pull request merged**